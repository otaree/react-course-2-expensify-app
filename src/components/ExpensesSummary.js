import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = (props) => (
    <div>
        <p>Viewing {props.expenses.length} expenses totaling {numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')}</p>
    </div>
);


const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesSummary); 