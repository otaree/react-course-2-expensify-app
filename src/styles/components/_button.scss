.button {
    background: $blue;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;
    line-height: 1;
    border: none; 
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888888;
    // margin-bottom: $s-size;
}