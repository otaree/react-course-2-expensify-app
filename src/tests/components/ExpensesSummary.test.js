import React from 'react';
import { shallow } from 'enzyme';
import numeral from 'numeral';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpressSummary correctly with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpressSummary correctly with expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});