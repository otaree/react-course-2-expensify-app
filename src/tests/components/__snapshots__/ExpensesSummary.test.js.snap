// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpressSummary correctly with expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totaling 
      <span>
        $232,323,232,324.24
      </span>
    </h1>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpressSummary correctly with no expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totaling 
      <span>
        $0.25
      </span>
    </h1>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
